src/extensions/surveys.tsx

```diff
@@ -219,3 +118,752 @@ export function generateSurveys(posthog: PostHog) {
+                            window.open(link)
+                        }
+                        onSubmit()
+                    }}
+                >
+                    {text}
+                </button>
+            </div>
+            {!appearance.whiteLabel && (
+                <PostHogLogo backgroundColor={appearance.backgroundColor || defaultBackgroundColor} />
+            )}
+        </div>
+    )
+}
+
+export function PostHogLogo({ backgroundColor }: { backgroundColor?: string }) {
+    const backgroundColorRef = Preact.createRef()
+    const [textColor, setTextColor] = useState('white')
+    useEffect(() => {
+        setTextColor(getTextColor(backgroundColorRef.current))
+    }, [])
+
+    return (
+        <a
+            href="https://posthog.com"
+            target="_blank"
+            rel="noopener"
+            ref={backgroundColorRef}
+            style={{ backgroundColor: backgroundColor, color: textColor }}
+            className="footer-branding auto-text-color"
+        >
+            Survey by {posthogLogo}
+        </a>
+    )
+}
+
+export function Surveys({ posthog, survey, style }: { posthog: PostHog; survey: Survey; style?: React.CSSProperties }) {
+    const [showConfirmation, setShowConfirmation] = useState(false)
+    const [showSurveyQuestion, setShowSurveyQuestion] = useState(true)
+
+    useEffect(() => {
+        window.dispatchEvent(new Event('PHSurveyShown'))
+        posthog.capture('survey shown', {
+            $survey_name: survey.name,
+            $survey_id: survey.id,
+            sessionRecordingUrl: posthog.get_session_replay_url?.(),
+        })
+        localStorage.setItem(`lastSeenSurveyDate`, new Date().toISOString())
+
+        window.addEventListener('PHSurveyClosed', () => {
+            setShowSurveyQuestion(false)
+        })
+
+        window.addEventListener('PHSurveySent', () => {
+            setShowSurveyQuestion(false)
+            setShowConfirmation(true)
+            if (survey.appearance?.autoDisappear) {
+                setTimeout(() => {
+                    setShowConfirmation(false)
+                }, 5000)
+            }
+        })
+    }, [])
+    const confirmationBoxLeftStyle = style?.left && _isNumber(style?.left) ? { left: style.left - 40 } : {}
+
+    return (
+        <>
+            {showSurveyQuestion && <Questions survey={survey} posthog={posthog} styleOverrides={style} />}
+            {survey.appearance?.displayThankYouMessage && showConfirmation && (
+                <ConfirmationMessage
+                    confirmationHeader={survey.appearance?.thankYouMessageHeader || 'Thank you!'}
+                    confirmationDescription={survey.appearance?.thankYouMessageDescription || ''}
+                    appearance={survey.appearance || {}}
+                    styleOverrides={{ ...style, ...confirmationBoxLeftStyle }}
+                    onClose={() => setShowConfirmation(false)}
+                />
+            )}
+        </>
+    )
+}
```

Comment:
There is a potential for bugs here. This is more of a simple state machine so you don't want multiple sources of truth but rather one option that decides what is shown. Something like: 

```suggestion
export function Surveys({ posthog, survey, style }: { posthog: PostHog; survey: Survey; style?: React.CSSProperties }) {
    const [displayState, setDisplayState] = useState<'survey' | 'confirmation' | 'closed'>('survey')

    useEffect(() => {
        window.dispatchEvent(new Event('PHSurveyShown'))
        posthog.capture('survey shown', {
            $survey_name: survey.name,
            $survey_id: survey.id,
            sessionRecordingUrl: posthog.get_session_replay_url?.(),
        })
        localStorage.setItem(`lastSeenSurveyDate`, new Date().toISOString())

        window.addEventListener('PHSurveyClosed', () => {
            setDisplayState('closed')
        })

        window.addEventListener('PHSurveySent', () => {
            setDisplayState('confirmation')
            if (survey.appearance?.autoDisappear) {
                setTimeout(() => {
                    setDisplayState('closed')
                }, 5000)
            }
        })
    }, [])
    const confirmationBoxLeftStyle = style?.left && _isNumber(style?.left) ? { left: style.left - 40 } : {}

    return (
        <>
            {displayState === 'survey' && <Questions survey={survey} posthog={posthog} styleOverrides={style} />}
            {displayState === 'confirmation' && survey.appearance?.displayThankYouMessage && (
                <ConfirmationMessage
                    confirmationHeader={survey.appearance?.thankYouMessageHeader || 'Thank you!'}
                    confirmationDescription={survey.appearance?.thankYouMessageDescription || ''}
                    appearance={survey.appearance || {}}
                    styleOverrides={{ ...style, ...confirmationBoxLeftStyle }}
                    onClose={() => setDisplayState('closed')}
                />
            )}
        </>
    )
}
```